pyccel/ast/core.py

```diff
@@ -1942,19 +1865,19 @@ def __init__(
         doc_string=None):
 
         if isinstance(name, str):
-            name = Symbol(name)
+            name = PyccelSymbol(name)
         elif isinstance(name, (tuple, list)):
             name_ = []
             for i in name:
                 if isinstance(i, str):
-                    name = name + Symbol(i)
-                elif not isinstance(i, Symbol):
-                    raise TypeError('Function name must be Symbol or string'
+                    name = name + PyccelSymbol(i)
```

Comment:
This code is also strange. Shouldn't it be `name_.append(PyccelSymbol(i))`?

Reply:
08d2e4f